src/main/java/training/endava/skeleton/Main.java

```diff
@@ -40,16 +47,90 @@ public static void main(String[] args) {
         personList.forEach(System.out::println);
         personRepository.deleteAll();
         personList.forEach(System.out::println);
+
+
+        List<Company> companies = createDummyDataCompany();
+        System.out.println("----------------Ex1------------------");
+        companies.forEach(System.out::println);
+        System.out.println("----------------Ex2------------------");
+        companies.stream().map(Company::getFoundationYear).mapToInt(Year::getValue).max().ifPresent(System.out::println);
+        System.out.println("-----------EX3-------------");
+        companies.stream().min((o1, o2) -> o1.getFoundationYear().compareTo(o2.getFoundationYear())).ifPresent(System.out::println);
+        System.out.println("-----------EX4-------------");
+        System.out.println("Number of persons: " + companies.stream().mapToInt(company -> company.getPersonList().size()).reduce(0, (o1, o2) -> o1 + o2));
+
+        System.out.println("-----------EX5-------------");
+        Person person1 = companies.stream()
+                .map(Company::getPersonList)
+                .flatMap(List<Person>::stream)
+                .reduce(new Person(), (o1, o2) -> {
+                    o2.setName(o1.getName() + o2.getName());
+                    return o2;
+                });
+
+        System.out.println(person1);
+        System.out.println("-----------EX6-------------");
+        companies.stream().map(Company::getPersonList)
+                .flatMap(List<Person>::stream)
+                .forEach(System.out::println);
+
+        System.out.println("----------EX7-------------");
+        companies.stream().map(Company::getPersonList)
+                .flatMap(List<Person>::stream).distinct().forEach(System.out::println);
+
+        System.out.println("----------EX8-------------");
+        Collector<Person, StringJoiner, String> personCollector =
+                Collector.of(
+                        () -> new StringJoiner(", ", "Prefix", "Suffix"),
+                        (supp, person2) -> supp.add(person2.getName().toUpperCase()),
+                        StringJoiner::merge,
+                        StringJoiner::toString
+                );
+
+        String names = companies.stream().map(Company::getPersonList)
+                .flatMap(List<Person>::stream).collect(personCollector);
+        System.out.println(names);
+
+        System.out.println("----------EX9-------------");
+        Instant startTime;
+        startTime = Instant.now();
+        Stream.iterate(0, i -> i + 1).limit(500000).filter(i -> i % 2 == 0).count();
+        System.out.println("NonParalel: " + Duration.between(Instant.now(), startTime).getSeconds());
```

Comment:
You'll get a negative value, you should swap the parameters, the first argument of between being the startTime

Reply:
The signature of between method is between(Temporal startInclusive, Temporal endExclusive) an it returns a positive number